src/runtime/components/ScrollArea.vue

```diff
@@ -0,0 +1,220 @@
+     */
+    gap?: number
+    /**
+     * Padding (in px) at the start of the list
+     * @defaultValue 0
+     */
+    paddingStart?: number
+    /**
+     * Padding (in px) at the end of the list
+     * @defaultValue 0
+     */
+    paddingEnd?: number
+    /**
+     * Scroll margin (in px) offset for scroll calculations
+     * @defaultValue 0
+     */
+    scrollMargin?: number
+    /**
+     * Number of lanes (columns for vertical, rows for horizontal)
+     * @defaultValue 1
+     */
+    lanes?: number
+    /**
+     * Custom key generation function for items
+     */
+    getItemKey?: (index: number) => string | number
+  }
+  class?: any
+  ui?: ScrollArea['slots']
+}
+
+export interface ScrollAreaSlots<T = any> {
+  default(props: { item: T, index: number, virtualItem?: VirtualItem } | Record<string, never>): any
+}
+</script>
+
+<script setup lang="ts" generic="T = any">
+import { computed, ref, toRef } from 'vue'
+import { Primitive } from 'reka-ui'
+import { defu } from 'defu'
+import { useVirtualizer } from '@tanstack/vue-virtual'
+import { useAppConfig } from '#imports'
+import { tv } from '../utils/tv'
+
+const props = withDefaults(defineProps<ScrollAreaProps<T>>(), {
+  as: 'div',
+  orientation: 'vertical',
+  virtualize: false
+})
+defineSlots<ScrollAreaSlots<T>>()
+
+const appConfig = useAppConfig() as ScrollArea['AppConfig']
+
+const rootRef = ref()
+
+const virtualizerProps = toRef(() => defu(typeof props.virtualize === 'boolean' ? {} : props.virtualize, {
+  estimateSize: 100,
+  overscan: 12,
+  gap: 0,
+  paddingStart: 0,
+  paddingEnd: 0,
+  scrollMargin: 0,
+  lanes: 1
+}))
+
+const virtualizer = useVirtualizer({
+  enabled: !!props.virtualize,
+  get count() {
+    return props.items?.length || 0
+  },
+  getScrollElement: () => (rootRef.value?.$el || null),
+  get horizontal() {
+    return props.orientation === 'horizontal'
+  },
+  get overscan() {
+    return virtualizerProps.value.overscan
+  },
+  estimateSize: () => virtualizerProps.value.estimateSize,
+  get gap() {
+    return virtualizerProps.value.gap
+  },
+  get paddingStart() {
+    return virtualizerProps.value.paddingStart
+  },
+  get paddingEnd() {
+    return virtualizerProps.value.paddingEnd
+  },
+  get scrollMargin() {
+    return virtualizerProps.value.scrollMargin
+  },
+  get lanes() {
+    return virtualizerProps.value.lanes
+  },
+  getItemKey: virtualizerProps.value.getItemKey
+})
+
+// Computed refs for virtual items and total size
+const virtualItems = computed(() => virtualizer.value.getVirtualItems())
+const totalSize = computed(() => virtualizer.value.getTotalSize())
+
+// Measure element for dynamic sizing
+function measureElement(el: Element | null) {
+  if (!el) return undefined
+  virtualizer.value.measureElement(el)
+  return undefined
+}
+
+const ui = computed(() => tv({ extend: tv(theme), ...(appConfig.ui?.scrollArea || {}) })({
+  orientation: props.orientation
+}))
+</script>
+
+<template>
+  <Primitive
```

Comment:
We need a way to not have this `root` slot when using in a component that already has one: https://github.com/nuxt/ui/blob/v4/src/runtime/components/Table.vue#L639